src/Robot.elm

```diff
@@ -0,0 +1,70 @@
+module Robot exposing (Model, Msg, init, subscriptions, update, view)
+
+import Browser
+import Browser.Events
+import Debug
+import Html
+import Html.Attributes as Attributes
+import Html.Events as Events exposing (onClick)
+import Json.Decode
+import String
+
+
+subscriptions : Model -> Sub Msg
+subscriptions _ =
+    Browser.Events.onKeyDown <| Json.Decode.map KeyPressed <| Json.Decode.field "key" Json.Decode.string
+
+
+type alias Model =
+    { x : Float
+    , y : Float
+    , up : String
+    , down : String
+    , left : String
+    , right : String
+    }
+
+
+type Msg
+    = KeyPressed String
+
+
+init : Float -> Float -> String -> String -> String -> String -> Model
+init x y up down right left =
+    { x = x, y = y, up = up, down = down, right = right, left = left }
+
+
+update : Msg -> Model -> Model
+update msg model =
+    let
+        _ =
+            Debug.log "msg" msg
+    in
+    case msg of
+        KeyPressed button ->
+            if button == model.up then
+                { model | y = model.y - 5 }
+
+            else if button == model.down then
+                { model | y = model.y + 5 }
+
+            else if button == model.right then
+                { model | x = model.x + 5 }
+
+            else if button == model.left then
+                { model | x = model.x - 5 }
```

Comment:
5 is a very random value. What if you will want someday to update this value? I assume it will be the same number in all those lines. Why not give it a name and a meaning?

Reply:
The "speed" should be received from outside, so that a robot can be initialized to drive at different speeds.
Perhaps even add a `Msg` that changes it's speed, and expose this functionality to the external user 🙂